--- a/com/mojang/authlib/BaseUserAuthentication.java
+++ b/com/mojang/authlib/BaseUserAuthentication.java
@@ -91,7 +91,8 @@
         this.userid = credentials.containsKey(STORAGE_KEY_USER_ID) ? String.valueOf(credentials.get(STORAGE_KEY_USER_ID)) : this.username;
         if (credentials.containsKey(STORAGE_KEY_USER_PROPERTIES)) {
             try {
-                List list = (List)credentials.get(STORAGE_KEY_USER_PROPERTIES);
+                // Weird patch to fix recompilation
+                List<Map> list = (List)credentials.get(STORAGE_KEY_USER_PROPERTIES);
                 for (Map propertyMap : list) {
                     String name = (String)propertyMap.get("name");
                     String value = (String)propertyMap.get("value");
@@ -111,7 +112,8 @@
             GameProfile profile = new GameProfile(UUIDTypeAdapter.fromString(String.valueOf(credentials.get(STORAGE_KEY_PROFILE_ID))), String.valueOf(credentials.get(STORAGE_KEY_PROFILE_NAME)));
             if (credentials.containsKey(STORAGE_KEY_PROFILE_PROPERTIES)) {
                 try {
-                    List list = (List)credentials.get(STORAGE_KEY_PROFILE_PROPERTIES);
+                    // Weird patch to fix recompilation
+                    List<Map> list = (List)credentials.get(STORAGE_KEY_PROFILE_PROPERTIES);
                     for (Map propertyMap : list) {
                         String name = (String)propertyMap.get("name");
                         String value = (String)propertyMap.get("value");
@@ -244,4 +246,3 @@
         this.userid = userid;
     }
 }
-
