--- a/net/minecraft/launcher/updater/CompleteMinecraftVersion.java
+++ b/net/minecraft/launcher/updater/CompleteMinecraftVersion.java
@@ -26,9 +26,12 @@
 import java.util.List;
 import java.util.Map;
 import java.util.Set;
+
+import me.nepnep.msa4legacy.patches.MicrosoftAccount;
 import net.minecraft.launcher.CompatibilityRule;
 import net.minecraft.launcher.CurrentLaunchFeatureMatcher;
 import net.minecraft.launcher.Launcher;
+import net.minecraft.launcher.profile.AuthenticationDatabase;
 import net.minecraft.launcher.profile.ProfileManager;
 import net.minecraft.launcher.updater.AbstractDownloadInfo;
 import net.minecraft.launcher.updater.Argument;
@@ -328,11 +331,14 @@
 
     public CompatibilityRule.FeatureMatcher createFeatureMatcher() {
         ProfileManager profileManager = Launcher.getCurrentInstance().getProfileManager();
-        UserAuthentication auth = profileManager.getAuthDatabase().getByUUID(profileManager.getSelectedUser());
-        return new CurrentLaunchFeatureMatcher(profileManager.getSelectedProfile(), this, auth);
+        AuthenticationDatabase authDb = profileManager.getAuthDatabase();
+        String selectedUser = profileManager.getSelectedUser();
+        UserAuthentication auth = authDb.getByUUID(selectedUser);
+        
+        return new CurrentLaunchFeatureMatcher(profileManager.getSelectedProfile(), this, auth, authDb.getMSAByUUID(selectedUser));
     }
 
-    public void addArguments(ArgumentType type, CompatibilityRule.FeatureMatcher featureMatcher, GameProcessBuilder builder, StrSubstitutor substitutor) {
+    public void addArguments(ArgumentType type, CompatibilityRule.FeatureMatcher featureMatcher, GameProcessBuilder builder, StrSubstitutor substitutor, MicrosoftAccount msa) {
         if (this.arguments != null) {
             List<Argument> args = this.arguments.get((Object)type);
             if (args != null) {
@@ -343,6 +349,9 @@
         } else if (this.minecraftArguments != null) {
             if (type == ArgumentType.GAME) {
                 for (String arg : this.minecraftArguments.split(" ")) {
+                    if (msa != null && (arg.equals("--userProperties") || arg.equals("${user_properties}"))) {
+                        continue;
+                    }
                     builder.withArguments(substitutor.replace(arg));
                 }
                 if (featureMatcher.hasFeature("is_demo_user", true)) {
@@ -369,4 +378,3 @@
         }
     }
 }
-
