--- a/net/minecraft/launcher/updater/MinecraftVersionManager.java
+++ b/net/minecraft/launcher/updater/MinecraftVersionManager.java
@@ -291,7 +291,7 @@
                 String filename = object.getHash().substring(0, 2) + "/" + object.getHash();
                 File file = new File(objectsFolder, filename);
                 if (file.isFile() && FileUtils.sizeOf(file) == object.getSize()) continue;
-                AssetDownloadable downloadable = new AssetDownloadable(proxy, entry.getValue(), object, "http://resources.download.minecraft.net/", objectsFolder);
+                AssetDownloadable downloadable = new AssetDownloadable(proxy, entry.getValue(), object, "https://resources.download.minecraft.net/", objectsFolder);
                 downloadable.setExpectedSize(object.getSize());
                 result.add(downloadable);
             }
@@ -300,14 +300,9 @@
             LOGGER.debug("Delta time to compare resources: " + delta / 1000000L + " ms ");
         }
         catch (Exception ex) {
-            try {
-                LOGGER.error("Couldn't download resources", (Throwable)ex);
-            }
-            catch (Throwable throwable) {
-                IOUtils.closeQuietly(inputStream);
-                throw throwable;
-            }
+            LOGGER.error("Couldn't download resources", (Throwable)ex);
             IOUtils.closeQuietly(inputStream);
+            throw new RuntimeException(ex);
         }
         IOUtils.closeQuietly(inputStream);
         return result;
@@ -362,4 +357,3 @@
         }
     }
 }
-
